src/transaction.rs

```diff
@@ -0,0 +1,13 @@
+pub trait Transaction {}
+
+pub enum TransactionSet<T> {
```

Comment:
This type seems overly complicated and doesn't properly restrict the input domain to valid structures. I'm worried about all the edge cases that may arise from handling this recursive structure. Can we just stick with a `Vec<Transaction>` for now?

Reply:
That is true. I replaced the transaction entirely by a `Payload` that will impose no restrictions to the contents of the block. https://github.com/FuelLabs/fuel-pbft/blob/76b73e81cc004ddb659ea19dcec47bd67307fcf6/src/block.rs#L8

Reply:
We now ask the network to generate a new payload at any given moment, without imposing restrictions:
https://github.com/FuelLabs/fuel-pbft/blob/76b73e81cc004ddb659ea19dcec47bd67307fcf6/src/network.rs#L33-L34